miasm2/ir/analysis.py

```diff
@@ -29,44 +28,43 @@ class ira_x86_16(ir_x86_16, ira)
 
     """
 
-    def call_effects(self, ad, instr):
-        """Default modelisation of a function call to @ad. This may be used to:
+    def call_effects(self, addr, instr):
+        """Default modelisation of a function call to @addr. This may be used to:
 
         * insert dependencies to arguments (stack base, registers, ...)
         * add some side effects (stack clean, return value, ...)
 
-        @ad: (Expr) address of the called function
+        @addr: (Expr) address of the called function
         @instr: native instruction which is responsible of the call
         """
 
         assignblk = AssignBlock({
-            self.ret_reg: ExprOp('call_func_ret', ad, self.sp),
-            self.sp: ExprOp('call_func_stack', ad, self.sp)},
+            self.ret_reg: ExprOp('call_func_ret', addr, self.sp),
+            self.sp: ExprOp('call_func_stack', addr, self.sp)},
             instr)
         return [assignblk]
 
-    def pre_add_instr(self, block, instr, irb_cur, ir_blocks_all, gen_pc_update):
+    def pre_add_instr(self, block, instr, assignments, ir_blocks_all, gen_pc_update):
         """Replace function call with corresponding call effects,
         inside the IR block"""
         if not instr.is_subcall():
-            return irb_cur
+            return False
         call_effects = self.call_effects(instr.args[0], instr)
-        for assignblk in call_effects:
-            irb_cur.irs.append(assignblk)
-        return None
+        assignments+= call_effects
+        return True
 
     def gen_equations(self):
         for irb in self.blocks.values():
             symbols_init = dict(self.arch.regs.all_regs_ids_init)
 
-            sb = SymbolicExecutionEngine(self, dict(symbols_init))
-            sb.emulbloc(irb)
+            symbexec = SymbolicExecutionEngine(self, dict(symbols_init))
+            symbexec.emulbloc(irb)
             eqs = []
-            for n_w in sb.symbols:
-                v = sb.symbols[n_w]
-                if n_w in symbols_init and symbols_init[n_w] == v:
+            for n_w in symbexec.symbols:
+                value = symbexec.symbols[n_w]
+                if n_w in symbols_init and symbols_init[n_w] == value:
                     continue
-                eqs.append(ExprAff(n_w, v))
+                eqs.append(ExprAff(n_w, value))
             print '*' * 40
             print irb
             irb.irs = [AssignBlock(eqs)]
```

Comment:
Here, the `irb.irs` is directly modified (now, it is r-only, right?). But this method seems to never be called... (and could actually have its main part replaced by `symbexec.to_assignblock()`)